Adds configs to disable laggy hard deletes (#59750)
tgstation/tgstation: config/config.txt

## Add the ID of the role you want assigning here
#DISCORD_ROLEID 000000000000000000

## Uncomment to disable hard deleting garbage collection failures. (Hard deleting GC failures causes lag in order to bring memory usage down and keep bugged objects from hanging around and causing bugs in other things.) You can safely enable this for performance on production if the lag spikes are too disruptive)
#DISABLE_GC_FAILURE_HARD_DELETES

## Uncomment to disable hard deletes entirely, even things that explicitly request. This is not recommended unless you have a need for it during events or other high pop times where performance is key.

> #DISABLE_ALL_HARD_DELETES
> 
> 
> 
> 
> 
> 
> 

Reviewer: ```suggestion
#DISABLE_ALL_HARD_DELETES

```

## Add the ID of the role you want assigning here
#DISCORD_ROLEID 000000000000000000

> ## Uncomment to disable hard deleting garbage collection failures. (Hard deleting GC failures causes lag in order to bring memory usage down and keep bugged objects from hanging around and causing bugs in other things.) You can safely enable this for preformance on production if the lag spikes are too disruptive)

Reviewer 2: ```suggestion
## Uncomment to disable hard deleting garbage collection failures. (Hard deleting GC failures causes lag in order to bring memory usage down and keep bugged objects from hanging around and causing bugs in other things.) You can safely enable this for performance on production if the lag spikes are too disruptive)
```

#DISABLE_GC_FAILURE_HARD_DELETES

> ## Uncommand to disable hard deletes entirely, even things that explicitly request. This is not recommended unless you have a need for it during events or other high pop times where preformance is key.

Reviewer 2: ```suggestion
## Uncomment to disable hard deletes entirely, even things that explicitly request. This is not recommended unless you have a need for it during events or other high pop times where performance is key.
```

## Add the ID of the role you want assigning here
#DISCORD_ROLEID 000000000000000000

## How long in seconds after which a hard delete is treated as causing lag. This can be a float and supports a precision as low as nanoseconds.
#HARD_DELETES_OVERRUN_THRESHOLD 0.5

## Once a typepath causes overrun from hard deletes this many times, stop hard deleting it on garbage collection failures. (set to 0 to disable)

> #HARD_DELETES_OVERRUN_LIMIT 0

Reviewer 3: the threshold needs to be enabled for the overrun limit to do anything doesn't it?

Author: yep. if threshold is set to 0, even the admin notices get disabled, and the if check for that overrun limit will never run.

the postpone code will still run.

Reviewer 3: It's a little unclear just from the config option, tho maybe that's better then giving people who won't read the code an understanding of it

Author: its 0.5 by default in code, hence the comment out in the config
